restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -616,74 +616,19 @@ public void deleteAtPath(
     db.delete(keyspace, collection, id, convertedPath, now);
   }
 
-  public JsonNode searchDocuments(
-      DocumentDB db,
-      String keyspace,
-      String collection,
-      String documentKey,
-      List<FilterCondition> filters,
-      List<PathSegment> path,
-      Boolean recurse,
-      int pageSize,
-      ByteBuffer pageState)
-      throws ExecutionException, InterruptedException, UnauthorizedException {
-    StringBuilder pathStr = new StringBuilder();
-
-    List<String> pathSegmentValues =
-        path.stream().map(PathSegment::getPath).collect(Collectors.toList());
-    List<Row> rows =
-        searchRows(
-                keyspace,
-                collection,
-                db,
-                Collections.emptyList(),
-                filters,
-                new ArrayList<>(),
-                pathSegmentValues,
-                recurse,
-                documentKey,
-                pageSize,
-                pageState)
-            .left;
-
-    if (rows.isEmpty()) return null;
-
-    ObjectNode docsResult = mapper.createObjectNode();
-    Map<String, List<Row>> rowsByDoc = new HashMap<>();
-    addRowsToMap(rowsByDoc, rows);
-
-    for (Map.Entry<String, List<Row>> entry : rowsByDoc.entrySet()) {
-      ImmutablePair<JsonNode, Map<String, List<JsonNode>>> result =
-          convertToJsonDoc(entry.getValue(), true, db.treatBooleansAsNumeric());
-      if (!result.right.isEmpty()) {
-        logger.info(String.format("Deleting %d dead leaves", result.right.size()));
-        db.deleteDeadLeaves(keyspace, collection, entry.getKey(), result.right);
-      }
-      JsonNode node = result.left.requiredAt(pathStr.toString());
-      docsResult.set(entry.getKey(), node);
-    }
-
-    if (docsResult.isMissingNode() || (docsResult.isObject() && docsResult.isEmpty())) {
-      return null;
-    }
-
-    return docsResult;
-  }
-
-  public ImmutablePair<JsonNode, ByteBuffer> searchDocumentsV2(
+  public JsonNode searchDocumentsV2(
```

Comment:
technically this should just be called `searchDocuments` from now on, but for the purposes of review I didn't want the diff to be illegible